### PR TITLE
MGMT-1719 Add vip_dhcp_allocation paramter to cluster and remove api_…

### PR DESCRIPTION
**Pull request by YuviGold**
…vip from terraform

### CODE REVIEW

File: discovery-infra/start_discovery.py [hunk 1]
```diff
@@ -469,6 +440,9 @@ def main():
         type=str,
         default='assisted-service'
     )
+    parser.add_argument(
+        "--skip-vip-dhcp-allocation", help="Disable VIP DHCP allocation mode", action="store_true"
```
Comment (tsorya) [inline discovery-infra/start_discovery.py:444]: what will happened if we skip dhcp allocation?

File: terraform_files/network/main.tf [hunk 2]
```diff
@@ -14,12 +14,5 @@ resource "libvirt_network" "net" {
 
   mtu = var.libvirt_network_mtu
 
-  dns {
```
Comment (ori-amizur) [inline terraform_files/network/main.tf:17]: Why are we removing previous support?  Shouldn't it be an additional option?

File: skipper.yaml [hunk 3]
```diff
@@ -35,7 +35,6 @@ env:
   ISO: $ISO
   SERVICE_BRANCH: $SERVICE_BRANCH
   SERVICE_REPO: $SERVICE_REPO
-  RUN_WITH_VIPS: $RUN_WITH_VIPS
```
Comment (ori-amizur) [inline skipper.yaml:38]: Why is this removed?

File: terraform_files/network/variables-libvirt.tf [hunk 4]
```diff
@@ -23,10 +23,6 @@ variable "cluster_domain" {
   description = "Cluster domain"
 }
 
-variable "api_vip" {
```
Comment (ori-amizur) [inline terraform_files/network/variables-libvirt.tf:26]: What about the ingress_vip?

File: discovery-infra/start_discovery.py [hunk 5]
```diff
@@ -128,27 +127,6 @@ def set_hosts_roles(client, cluster_id, network_name):
     client.set_hosts_roles(cluster_id=cluster_id, hosts_with_roles=added_hosts)
 
 
-def set_cluster_vips(client, cluster_id):
```
Comment (ori-amizur) [inline discovery-infra/start_discovery.py:131]: Why remove?
Where do we set the machine cidr?

File: skipper.yaml (hunk 3 above)
Comment (YuviGold) [reply skipper.yaml:38]: It wasn't doing anything

File: discovery-infra/start_discovery.py (hunk 1 above)
Comment (YuviGold) [reply discovery-infra/start_discovery.py:470]: Same old VIPs allocation

File: terraform_files/network/variables-libvirt.tf (hunk 4 above)
Comment (YuviGold) [reply terraform_files/network/variables-libvirt.tf]: It wasn't assigned before @tsorya

**Comment by ori-amizur** [comment]
@YuviGold  Where do we set machine-cidr ?

**Comment by YuviGold** [comment]
> @YuviGold Where do we set machine-cidr ?

It is already being taken care of by test-infra.
```python
    parser.add_argument(
        "-vN",
        "--vm-network-cidr",
        help="Vm network cidr",
        type=str,
        default="192.168.126.0/24",
    )
...
# Converts params from args to assisted-service cluster params
def _cluster_create_params():
    params = {
        "openshift_version": args.openshift_version,
        "base_dns_domain": args.base_dns_domain,
        "cluster_network_cidr": args.cluster_network,
        "cluster_network_host_prefix": args.host_prefix,
        "service_network_cidr": args.service_network,
        "pull_secret": args.pull_secret,
        "http_proxy": args.http_proxy,
        "https_proxy": args.https_proxy,
        "no_proxy": args.no_proxy,
        "vip_dhcp_allocation": not args.skip_vip_dhcp_allocation,
    }
    return params

# convert params from args to terraform tfvars
def _create_node_details(cluster_name):
    return {
        "libvirt_worker_memory": args.worker_memory,
        "libvirt_master_memory": args.master_memory,
        "worker_count": args.number_of_workers,
        "cluster_name": cluster_name,
        "cluster_domain": args.base_dns_domain,
        "machine_cidr": args.vm_network_cidr,
        "libvirt_network_name": args.network_name,
        "libvirt_network_mtu": args.network_mtu,
        "libvirt_network_if": args.network_bridge,
        "libvirt_worker_disk": args.worker_disk,
        "libvirt_master_disk": args.master_disk,
    }
```

**Comment by ori-amizur** [comment]
> > @YuviGold Where do we set machine-cidr ?
> 
> It is already being taken care of by test-infra.
> 
> ```python
>     parser.add_argument(
>         "-vN",
>         "--vm-network-cidr",
>         help="Vm network cidr",
>         type=str,
>         default="192.168.126.0/24",
>     )
> ...
> # Converts params from args to assisted-service cluster params
> def _cluster_create_params():
>     params = {
>         "openshift_version": args.openshift_version,
>         "base_dns_domain": args.base_dns_domain,
>         "cluster_network_cidr": args.cluster_network,
>         "cluster_network_host_prefix": args.host_prefix,
>         "service_network_cidr": args.service_network,
>         "pull_secret": args.pull_secret,
>         "http_proxy": args.http_proxy,
>         "https_proxy": args.https_proxy,
>         "no_proxy": args.no_proxy,
>         "vip_dhcp_allocation": not args.skip_vip_dhcp_allocation,
>     }
>     return params
> 
> # convert params from args to terraform tfvars
> def _create_node_details(cluster_name):
>     return {
>         "libvirt_worker_memory": args.worker_memory,
>         "libvirt_master_memory": args.master_memory,
>         "worker_count": args.number_of_workers,
>         "cluster_name": cluster_name,
>         "cluster_domain": args.base_dns_domain,
>         "machine_cidr": args.vm_network_cidr,
>         "libvirt_network_name": args.network_name,
>         "libvirt_network_mtu": args.network_mtu,
>         "libvirt_network_if": args.network_bridge,
>         "libvirt_worker_disk": args.worker_disk,
>         "libvirt_master_disk": args.master_disk,
>     }
> ```

This is not enough, because you need to set it in the UpdateCluster operation.  Before, we set the vips. In DHCP mode we need to set the machine-cidr.  This is a new parameter to UpdateCluster that did not exist before.

**Comment by YuviGold** [comment]
> ```python
> cluster_network_cidr
> ```

Ok I thought `cluster_network_cidr` is like `machine_network_cidr`. Maybe it should be a part of the `cluster-create-params`?

**Comment by ori-amizur** [comment]
> > ```python
> > cluster_network_cidr
> > ```
> 
> Ok I thought `cluster_network_cidr` is like `machine_network_cidr`. Maybe it should be a part of the `cluster-create-params`?

No because of the following reasons:
1. The machine-cidr has to be one of the host-networks,  The host-networks are generated from the inventories that are received from the hosts.  They do not exist when cluster is registered.
2. In case of multiple host-networks we should allow the user to move from one network to another.  This has to be done in the UpdateCluster operation.

**Review by ori-amizur — APPROVED** [review]

File: discovery-infra/start_discovery.py [hunk 6]
```diff
@@ -300,6 +313,17 @@ def main():
         nodes_flow(client, cluster_name, cluster)
 
 
+def str2bool(v):
```
Comment (tsorya) [inline discovery-infra/start_discovery.py:316]: what is it? just pass 'y' or 'n' , u handle it in Makefile why u need to have conversions? just pass relevant string

File: Makefile [hunk 7]
```diff
@@ -27,6 +27,7 @@ OPENSHIFT_VERSION := $(or $(OPENSHIFT_VERSION), 4.5)
 REMOTE_SERVICE_URL := $(or $(REMOTE_SERVICE_URL), "")
 WORKER_DISK ?= 21474836480
 MASTER_DISK ?= 21474836480
+VIP_DHCP_ALLOCATION := $(or $(VIP_DHCP_ALLOCATION),yes)
```
Comment (tsorya) [inline Makefile:30]: why it is not in # network params ?

**Review by tsorya — CHANGES_REQUESTED** [review]
no need for boolean conversion func

File: discovery-infra/start_discovery.py (hunk 6 above)
Comment (YuviGold) [reply discovery-infra/start_discovery.py:316]: It is more capable :)

Comment (tsorya) [reply discovery-infra/start_discovery.py:316]: it is not

Comment (tsorya) [reply discovery-infra/start_discovery.py:316]: use at least distutils.util.strtobool

Comment (tsorya) [inline discovery-infra/start_discovery.py:316]: it is not more capable :) 
use at least distutils.util.strtobool
but i think it is not needed
